flag.go

```diff
@@ -174,18 +176,34 @@ func sortFlags(flags map[normalizedName]*Flag) []*Flag {
 	list.Sort()
 	result := make([]*Flag, len(list))
 	for i, name := range list {
-		result[i] = flags[normalizedName(name)]
+		result[i] = flags[NormalizedName(name)]
 	}
 	return result
 }
 
-func (f *FlagSet) normalizeFlagName(name string) normalizedName {
+func NormalizeWordSeparators(f *FlagSet, name string) NormalizedName {
 	result := name
 	for _, sep := range f.wordSeparators {
 		result = strings.Replace(result, sep, "-", -1)
 	}
 	// Type convert to indicate normalization has been done.
-	return normalizedName(result)
+	return NormalizedName(result)
+}
+
+func (f *FlagSet) SetNormalizeName(n func(f *FlagSet, name string) NormalizedName) {
+	f.normalizeNameFunc = n
```

Comment:
This suffers the same bug as my #13 fixed - if I set the normalize function after some flags have been set, they do not get normalized.  You have to loop through all existing flags in the set and normalize them first.